cumulus/parachains/runtimes/assets/asset-hub-rococo/src/lib.rs

```diff
@@ -1799,6 +1799,18 @@ impl_runtime_apis! {
 	}
 }
 
+fn get_assets_in_pool_with(asset: xcm::v4::Location) -> impl Iterator<Item = AssetId> {
```

Comment:
I think just reference is enough here:
```suggestion
fn get_assets_in_pool_with(asset: &xcm::v4::Location) -> impl Iterator<Item = AssetId> {
```
then you don't need to clone all the time: `get_assets_in_pool_with(native_token.clone());` but just `get_assets_in_pool_with(&native_token);`